hugegraph-llm/src/hugegraph_llm/operators/llm_op/schema_build.py

```
@@ -0,0 +1,117 @@
+
+    def _extract_schema(self, response: str) -> Dict[str, Any]:
+        # Try to extract JSON from Markdown code block
+        match = re.search(r"```(?:json)?\s*(.*?)```", response, re.DOTALL)
+        if match:
+            response = match.group(1).strip()
+
+        try:
+            return json.loads(response)
+        except json.JSONDecodeError as e:
+            log.error("Failed to parse LLM response as JSON: %s", response)
+            raise RuntimeError(f"Invalid JSON response from LLM: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)